File: bin/backup.rb
#!/usr/bin/env ruby

def find_backup_drive
  %w(home-silver home-gold home-rose SAGAN).each do |disk|
    if File.exists? "/Volumes/#{disk}"
      return disk
    end
  end

  puts 'Please insert your backup drive'
  exit 1
end

drive = find_backup_drive
backup_dir = "/Volumes/#{drive}/home"
puts "Backing up to #{backup_dir}..."

dirs = %w(books code Compositions Documents dotfiles Movies Music Pictures thoughts)

if !Dir.exists?(backup_dir)
  puts "Please create directory '#{backup_dir}'"
  exit 1
end

Dir.chdir(Dir.home) do
  dirs.each do |dir|
    if !Dir.exists?(dir)
      puts "~/#{dir} doesn't exist"
      next
    end

    puts '*' * 80
    puts "  backing up #{dir}..."
    puts '*' * 80

    # -vr: verbose, recursive
    # --size-only: don't copy files if dest and source are the same size
    # --delete: delete files in dest that aren't in source
    system("rsync -vr --size-only --delete #{Dir.home}/#{dir}/ #{backup_dir}/#{dir}")
  end

  system("cp ~/Library/Application\\ Support/Google/Chrome/Default/Bookmarks #{backup_dir}/bookmarks.json")
end

File: bin/.backup.py
#!/usr/bin/env python3

import os

def find_backup_drive():
    for disk in ['home-silver', 'home-gold', 'home-rose', 'SAGAN']:
        if os.path.exists(f'/Volumes/{disk}'):
            return disk
    print('Please insert your backup drive')
    exit(1)

drive = find_backup_drive()
backup_dir = f'/Volumes/{drive}/home'
print(f'Backing up to {backup_dir}...')

dirs = ['books', 'code', 'Compositions', 'Documents', 'dotfiles', 'Movies', 'Music', 'Pictures', 'thoughts']

if not os.path.exists(backup_dir):
    print(f"Please create directory '{backup_dir}'")
    exit(1)

home = os.path.expanduser('~')
os.chdir(home)

for dir in dirs:
    if not os.path.exists(dir):
        print(f"~/{dir} doesn't exist")
        continue

    print('*' * 80)
    print(f'  backing up {dir}...')
    print('*' * 80)

    # -vr: verbose, recursive
    # --size-only: don't copy files if dest and source are the same size
    # --delete: delete files in dest that aren't in source
    os.system(f'rsync -vr --size-only --delete {home}/{dir}/ {backup_dir}/{dir}')

print('\n  __________,\n /  Done!  /\n\'---------')
